ui/src/app/views/new-notebook-modal/component.ts

```diff
@@ -0,0 +1,78 @@
+import {Component, Input, OnDestroy} from '@angular/core';
+
+import {SignInService} from 'app/services/sign-in.service';
+import {Kernels} from 'app/views/notebook-redirect/component';
+
+import {environment} from 'environments/environment';
+
+import {Workspace} from 'generated';
+
+
+@Component({
+  selector: 'app-new-notebook-modal',
+  styleUrls: ['./component.css',
+    '../../styles/buttons.css',
+    '../../styles/inputs.css'],
+  templateUrl: './component.html',
+})
+export class NewNotebookModalComponent implements OnDestroy {
+  public creatingNotebook = false;
+  public newName = '';
+  @Input() workspace: Workspace;
+  Kernels = Kernels;
+  kernelType: number = Kernels.Python3;
+  notebookAuthListeners: EventListenerOrEventListenerObject[] = [];
+
+  loading = false;
+
+  constructor(
+    private signInService: SignInService,
+  ) {}
+
+  ngOnDestroy(): void {
+    this.notebookAuthListeners.forEach(f => window.removeEventListener('message', f));
+  }
+
+  open(): void {
+    this.creatingNotebook = true;
+    this.loading = false;
+  }
+
+  close(): void {
+    this.creatingNotebook = false;
+  }
+
+  submitNewNotebook(): void {
+    const nbUrl = `/workspaces/${this.workspace.namespace} \
+        ${this.workspace.id}/notebooks/create/${this.newName}/${this.kernelType}`;
```

Comment:
You'll need to encode the notebook name: https://github.com/all-of-us/workbench/blob/1956c20e0319bffe1a7e0c7386d6542daffba233/ui/src/app/views/notebook-list/component.ts#L133

Reply:
Will do.